### DISCUSSION TITLE
Compiling a model from cfg / meta / weights file into .so file

### DISCUSSION THREAD

**Discussion by werdnum** [AlexeyAB/darknet#7551] [Q&A]
Hi folks,

I'm trying to set up [this project](https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/) to run on my Jetson Nano device. The code seems to [ship with `.so` files](https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/tree/master/ml_api/bin), but they're linked against OpenCV 3.3 so I can't use them on my device, which has OpenCV 4.X installed - I know this because when I start up my container, I get this error:

So I'm trying to produce new `.so` files using `darknet` - but honestly I haven't really had much luck in finding comprehensive documentation that covers all the options available for the `darknet` binary - there doesn't seem to be a `darknet help` command or a manual page or anything. It looks like `build.sh` is there to build darknet itself, but I want to compile a whole *model* as a `.so` file, unless somehow I'm misunderstanding what [these files](https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/tree/master/ml_api/bin) are for.

The code I'm using does seem to [ship with](https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/tree/master/ml_api/model) some files that seem like the inputs that `darknet` wants, but there aren't any instructions there either for turning those files into the `.so` files that I guess are there to compile the model into a server (or dynamically link it in or whatever).

Any advice / pointers greatly appreciated for figuring out how to do this.

**Comment by stephanecharette** [comment]
Since you're having problems with that repo...why don't you ask them instead?  It is unlikely you'll find help in this repo about a different one.

There is no help command, but the command line options are described in the readme.

**Reply by werdnum** [reply]
Fair question, I've posted [an issue](https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/issues/414#issuecomment-808865064) over there. I wondered if those `.so` files were perhaps compiled by a generic mechanism in `darknet`.

Is [this](https://github.com/AlexeyAB/darknet#how-to-use-on-the-command-line) the documentation you are referring to?

**Reply by stephanecharette** [reply]
Correct, the readme.md file.  Another place you can get help is the discord https://discord.gg/zSq8rtW as well as the FAQ https://www.ccoderun.ca/programming/darknet_faq/
